main.go

```diff
@@ -666,76 +950,163 @@ is available in the $BITRISE_IDEDISTRIBUTION_LOGS_PATH environment variable`)
 
 	// Export .ipa
 	fmt.Println()
 
-	if err := utils.ExportOutputFile(ipaPath, ipaPath, bitriseIPAPthEnvKey); err != nil {
-		fail("Failed to export %s, error: %s", bitriseIPAPthEnvKey, err)
+	if err := utils.ExportOutputFile(opts.IPAPath, opts.IPAPath, bitriseIPAPthEnvKey); err != nil {
+		return fmt.Errorf("failed to export %s, error: %s", bitriseIPAPthEnvKey, err)
 	}
 
-	log.Donef("The ipa path is now available in the Environment Variable: %s (value: %s)", bitriseIPAPthEnvKey, ipaPath)
+	log.Donef("The ipa path is now available in the Environment Variable: %s (value: %s)", bitriseIPAPthEnvKey, opts.IPAPath)
 
 	// Export .dSYMs
 	fmt.Println()
 
 	appDSYM, frameworkDSYMs, err := archive.FindDSYMs()
 	if err != nil {
-		fail("Failed to export dsyms, error: %s", err)
+		return fmt.Errorf("failed to export dsyms, error: %s", err)
 	}
 
 	if err == nil {
 		dsymDir, err := pathutil.NormalizedOSTempDirPath("__dsyms__")
 		if err != nil {
-			fail("Failed to create tmp dir, error: %s", err)
+			return fmt.Errorf("failed to create tmp dir, error: %s", err)
 		}
 
 		if len(appDSYM) > 0 {
 			if err := exportDSYMs(dsymDir, appDSYM); err != nil {
-				fail("Failed to export dSYMs: %v", err)
+				return fmt.Errorf("failed to export dSYMs: %v", err)
 			}
 		} else {
 			log.Warnf("no app dsyms found")
 		}
 
-		if cfg.ExportAllDsyms {
+		if opts.ExportAllDsyms {
 			if err := exportDSYMs(dsymDir, frameworkDSYMs); err != nil {
-				fail("Failed to export dSYMs: %v", err)
+				return fmt.Errorf("failed to export dSYMs: %v", err)
 			}
 		}
 
 		if err := utils.ExportOutputDir(dsymDir, dsymDir, bitriseDSYMDirPthEnvKey); err != nil {
-			fail("Failed to export %s, error: %s", bitriseDSYMDirPthEnvKey, err)
+			return fmt.Errorf("failed to export %s, error: %s", bitriseDSYMDirPthEnvKey, err)
 		}
 
 		log.Donef("The dSYM dir path is now available in the Environment Variable: %s (value: %s)", bitriseDSYMDirPthEnvKey, dsymDir)
 
-		if err := utils.ExportOutputDirAsZip(dsymDir, dsymZipPath, bitriseDSYMPthEnvKey); err != nil {
-			fail("Failed to export %s, error: %s", bitriseDSYMPthEnvKey, err)
+		if err := utils.ExportOutputDirAsZip(dsymDir, opts.DSYMZipPath, bitriseDSYMPthEnvKey); err != nil {
+			return fmt.Errorf("failed to export %s, error: %s", bitriseDSYMPthEnvKey, err)
 		}
 
-		log.Donef("The dSYM zip path is now available in the Environment Variable: %s (value: %s)", bitriseDSYMPthEnvKey, dsymZipPath)
+		log.Donef("The dSYM zip path is now available in the Environment Variable: %s (value: %s)", bitriseDSYMPthEnvKey, opts.DSYMZipPath)
+	}
+
+	return nil
+}
+
+// RunStep ...
+func RunStep() error {
+	step := NewXcodeArchiveStep()
+
+	config, err := step.ProcessInputs()
+	if err != nil {
+		return err
+	}
+
+	installDepsOpts := InstallDepsOpts{
+		InstallXcpretty: config.OutputTool == "xcpretty",
+	}
+	if err := step.InstallDeps(installDepsOpts); err != nil {
+		return err
+	}
+
+	runOpts := RunOpts{
```

Comment:
NOTE: I find it counter-intiutive that step.Run has no "outputs", only by side effects of placing files on predefined paths.